### PR TITLE
EZEE-3167: Added generation of Site Access name for Sites

### PR DESCRIPTION
**Pull request by mikadamczyk**
| Question      | Answer
| ------------- | ---
| Tickets       | https://jira.ez.no/browse/EZEE-3167
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Tests pass?   | yes
| Doc needed?   | no
| Related PR     | https://github.com/ezsystems/ezplatform-site-factory/pull/70
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

List of the Site Accesses should come from `siteAccessService`. Site Factory should have the possibility to modify the Site Access name.

#### Checklist:
- [x] Coding standards (`$ composer fix-cs`)
- [x] Ready for Code Review

### CODE REVIEW

File: src/lib/Siteaccess/NonAdminSiteaccessResolver.php [hunk 1]
```diff
@@ -42,6 +42,29 @@ public function getSiteaccessesForLocation(
...
+                'siteaccess',
+                'SiteAccess group `admin_group` not found.'
+            );
+        }
+
+        return array_filter(
+            $this->siteaccessResolver->getSiteAccessesListForLocation($location, $versionNo, $languageCode),
+            function ($siteAccess) {
+                return !in_array($siteAccess->name, $this->siteAccessGroups['admin_group']);
+            }
```
Comment (adamwojs) [inline src/lib/Siteaccess/NonAdminSiteaccessResolver.php:64]: Please reuse `\EzSystems\EzPlatformAdminUi\Specification\SiteAccess\IsAdmin` specification here.

File: src/lib/Siteaccess/SiteaccessResolver.php [hunk 2]
```diff
@@ -51,27 +52,53 @@ public function getSiteaccessesForLocation(
...
     public function getSiteaccesses(): array
     {
-        return $this->siteAccesses;
+        return $this->getSiteAccessList();
+    }
+
+    /**
+     * @return string[]
+     */
+    private function getSiteAccessList(): array
```
Comment (adamwojs) [inline src/lib/Siteaccess/SiteaccessResolver.php:97]: Please add SA list as an argument and reuse it in `getSiteaccesses` and `getSiteaccessesForLocation` (we will avoid `array_column($saList, 'name')` duplication)

File: src/bundle/Resources/views/themes/admin/content/content_preview.html.twig [hunk 3]
```diff
@@ -41,8 +41,20 @@
...
+                                    {
+                                        'contentId': content.id,
+                                        'versionNo': version_no,
+                                        'language': language_code,
+                                        'siteAccessName': site_accesses_name
+                                    }
+                                )
+                            }}">
+                                {{ site_access_label|trans({}, 'ezplatform_siteaccess') }}
+                            </option>
```
Comment (adamwojs) [inline src/bundle/Resources/views/themes/admin/content/content_preview.html.twig:57]: Introducing variable might improve readability even more :wink: 

```twig
{% set option_value = path(
    '_ezpublishPreviewContent',
    {
        'contentId': content.id,
        'versionNo': version_no,
        'language': language_code,
        'siteAccessName': site_accesses_name
    }
) %}

<option value="{{ option_value }}">
    {{ site_access_label|trans({}, 'ezplatform_siteaccess') }}
</option>
```

File: src/lib/Siteaccess/SiteaccessResolverInterface.php [hunk 4]
```diff
@@ -28,7 +28,18 @@ public function getSiteaccessesForLocation(
...
 
     /**
-     * Returns complete list of siteaccesses.
+     * Returns a complete list of Site Access objects.
+     *
+     * @return \eZ\Publish\Core\MVC\Symfony\SiteAccess[]
+     */
+    public function getSiteAccessesListForLocation(
+        Location $location,
+        int $versionNo = null,
```
Comment (ViniTou) [inline src/lib/Siteaccess/SiteaccessResolverInterface.php:37]: both should be nullable

```suggestion
        ?int $versionNo = null,
```

File: src/lib/Siteaccess/SiteaccessResolverInterface.php [hunk 5]
```diff
@@ -28,7 +28,18 @@ public function getSiteaccessesForLocation(
...
+        string $languageCode = null
```
Comment (ViniTou) [inline src/lib/Siteaccess/SiteaccessResolverInterface.php:38]:
```suggestion
        ?string $languageCode = null
```

Comment (mikadamczyk) [reply src/lib/Siteaccess/SiteaccessResolverInterface.php]: @ViniTou fixed in https://github.com/ezsystems/ezplatform-admin-ui/pull/1405/files/901eb9340f62841d61143946651d119acee43f81..1faec98f10e0dd188cbe13b8816146fa7acf13eb

**Comment by lserwatka** [comment]
You can merge it up.

**Pull request merged**